Sources/KlaviyoUI/KlaviyoWebView/Development Assets/JSTestWebViewModel.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  JSTestWebViewModel.swift
+//  klaviyo-swift-sdk
+//
+//  Created by Andrew Balmer on 11/18/24.
+//
+
+#if DEBUG
+import Combine
+import Foundation
+import WebKit
+
+class JSTestWebViewModel: KlaviyoWebViewModeling {
+    let url: URL
+    let loadScripts: [String: WKUserScript]?
+
+    /// Publishes scripts for the `WKWebView` to execute.
+    private var continuation: AsyncStream<(script: String, callback: ((Result<Any?, Error>) -> Void)?)>.Continuation?
+    lazy var scriptStream: AsyncStream<(script: String, callback: ((Result<Any?, Error>) -> Void)?)> = AsyncStream { [weak self] continuation in
+        self?.continuation = continuation
+    }
+
+    init(url: URL) {
+        self.url = url
+        loadScripts = JSTestWebViewModel.initializeLoadScripts()
+    }
+
+    private static func initializeLoadScripts() -> [String: WKUserScript] {
+        var scripts: [String: WKUserScript] = [:]
+
+        if let toggleHandlerScript = try? FileIO.getFileContents(path: "toggleHandler", type: "js") {
+            let script = WKUserScript(source: toggleHandlerScript, injectionTime: .atDocumentEnd, forMainFrameOnly: true)
+            scripts["toggleMessageHandler"] = script
+        }
+
+        return scripts
+    }
+
+    // MARK: handle WKWebView events
+
+    func handleNavigationEvent(_ event: WKNavigationEvent) {
+        // TODO: handle navigation events
```

Comment:
what navigation events?

Reply:
the ViewController currently passes navigation events to the ViewModel (see [here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/cb8185901e00bb1c2997a91010eee6361bed9b51/Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebViewController.swift#L108)). These events can be used to let us know when the website loads or if the website fails to load. 

We may not end up using this for the `JSTestWebViewModel`, but I can leave it here for now in case we decide to do some testing based on navigation events.